pkg/watcher/cooldownqueue/cooldownqueue.go

```diff
@@ -40,11 +41,10 @@ func NewCooldownQueue() *CooldownQueue {
 
 // makeEventKey creates a unique key for an event from a watcher
 func makeEventKey(e watch.Event) string {
-	object, ok := e.Object.(*unstructured.Unstructured)
-	if !ok {
-		return ""
-	}
-	return strings.Join([]string{object.GroupVersionKind().Group, object.GroupVersionKind().Version, object.GetKind(), object.GetNamespace(), object.GetName()}, "/")
+	object := e.Object.(runtime.Object)
```

Comment:
we'll get a panic if the assertion fails (which is better than returning "")

Reply:
Do we really want to panic in production (?)

Reply:
yes, but we are catching the missing resource in other places before reaching that